src/main/java/core/basesyntax/service/impl/FruitTransactionServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.FruitTransactionService;
+import core.basesyntax.service.ReaderService;
+import java.util.ArrayList;
+import java.util.List;
+
+public class FruitTransactionServiceImpl implements FruitTransactionService {
+    private static final String TITLE = "type";
+    private static final String SEPARATOR = ",";
+    private static final int FRUIT_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+    private static final int OPERATION_INDEX = 0;
+    private static final int TRANSACTION_DATA_QUANTITY = 3;
+
+    private final ReaderService readerService = new ReaderServiceImpl();
+
+    @Override
+    public List<FruitTransaction> getFruitTransactionsFromCvsFile(String fileName) {
+        List<String> fruitInfoFromFile = readerService.getInformationFromCvsFile(fileName);
+        List<FruitTransaction> fruitTransactions = new ArrayList<>();
+        for (String information : fruitInfoFromFile) {
+            String[] infoAboutFruitTransaction = information.split(SEPARATOR);
+            if (infoAboutFruitTransaction[0].equals(TITLE)) {
+                continue;
+            }
+            validateTransactionInfo(information);
+            FruitTransaction fruitTransaction = new FruitTransaction(getOperation(
+                    infoAboutFruitTransaction[OPERATION_INDEX]),
+                            infoAboutFruitTransaction[FRUIT_INDEX],
+                            Integer.parseInt(infoAboutFruitTransaction[QUANTITY_INDEX]));

```

Comment:
Such structure is not readable. Please prepare some variable before with part of parsing or extract to separate method to improve readability